--- a/sshfs.c
+++ b/sshfs.c
@@ -1081,27 +1081,29 @@
 		return -EIO;
 
 	for (i = 0; i < count; i++) {
-		int err = -1;
+		int err;
 		char *name;
 		char *longname;
 		struct stat stbuf;
 		if (buf_get_string(buf, &name) == -1)
 			return -EIO;
-		if (buf_get_string(buf, &longname) != -1) {
-			free(longname);
-			err = buf_get_attrs(buf, &stbuf, NULL);
-			if (!err) {
-				struct dir_entry *entry;
-				if (sshfs.follow_symlinks &&
-				    S_ISLNK(stbuf.st_mode)) {
-					stbuf.st_mode = 0;
-				}
-				entry = g_new(struct dir_entry, 1);
-				entry->name = name;
-				entry->stbuf = stbuf;
-				g_ptr_array_add(entries, entry);
-				name = NULL; /* ownership passed to entry */
+		if (buf_get_string(buf, &longname) == -1) {
+			free(name);
+			return -EIO;
+		}
+		free(longname);
+		err = buf_get_attrs(buf, &stbuf, NULL);
+		if (!err) {
+			struct dir_entry *entry;
+			if (sshfs.follow_symlinks &&
+			    S_ISLNK(stbuf.st_mode)) {
+				stbuf.st_mode = 0;
 			}
+			entry = g_new(struct dir_entry, 1);
+			entry->name = name;
+			entry->stbuf = stbuf;
+			g_ptr_array_add(entries, entry);
+			name = NULL; /* ownership passed to entry */
 		}
 		free(name);
 		if (err)
